### PR TITLE
Origin ref based date calculator (RWA 1966)

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

[RWA-1966](https://tools.hmcts.net/jira/browse/RWA-1966)

### Change description ###

Origin ref based date calculator

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+import java.util.List;
+
+import static uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.calendar.DateTypeIntervalData.DATE_TYPE_MUST_BE_WORKING_DAY_NEXT;
+import static uk.gov.hmcts.reform.wataskmanagementapi.services.calendar.PublicHolidaysCollectionTest.CALENDAR_URI;
+
+@SpringBootTest
+@ActiveProfiles({"integration"})
+public class OriginRefDateTypeConfiguratorTest {
+
+    public static final LocalDateTime GIVEN_DATE = LocalDateTime.of(2022, 10, 13, 18, 00, 00);
```
Comment (chrisdavecm) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java:23]: This is only summer time do we need to test winter date as well?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+    public static final LocalDateTime BST_DATE_BACKWARD = LocalDateTime.of(2022, 10, 26, 18, 00, 00);
```
Comment (chrisdavecm) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java:25]: These aren't used

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+                    .name(CamundaValue.stringValue("calculatedDates"))
+                    .value(CamundaValue.stringValue("nextHearingDate,dueDate,priorityDate"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(CamundaValue.stringValue("nextHearingDate"))
+                    .value(CamundaValue.stringValue("2022-10-13T16:00"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(CamundaValue.stringValue("dueDate"))
+                    .value(CamundaValue.stringValue("2022-10-16T17:00"))
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java:80]: For nexteHearingDate = "2022-10-13T16:00", dueDateintervalDays = 3, dueDateSkipNonWorkingDays = true,

dueDate is calculated as "2022-10-16T17:00", should this be 2022-10-17T17:00"?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(CamundaValue.stringValue("priorityDate"))
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/OriginRefDateTypeConfiguratorTest.java:83]: For nextHearingDate = "2022-10-13T16:00", priorityDateintervalDays = 3, priorityDateSkipNonWorkingDays = true,

priorityDate is calculated as "2022-10-22T21:00", should this be 2022-10-17T17:00"?

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**